x-pack/plugins/infra/public/components/asset_details/tabs/overview/overview.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiCallOut, EuiFlexGroup, EuiFlexItem, EuiLink } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { DataView } from '@kbn/data-views-plugin/public';
+import type { StringDateRange } from '../../../../pages/metrics/hosts/hooks/use_unified_search_url_state';
+import type { InventoryItemType } from '../../../../../common/inventory_models/types';
+import { findInventoryModel } from '../../../../../common/inventory_models';
+import type { MetricsTimeInput } from '../../../../pages/metrics/metric_detail/hooks/use_metrics_time';
+import { useMetadata } from '../../hooks/use_metadata';
+import { useSourceContext } from '../../../../containers/metrics_source';
+import { MetadataSummary } from './metadata_summary';
+import { KPIGrid } from './kpi_grid';
+import type { TabIds } from '../../types';
+
+export interface MetadataSearchUrlState {
+  metadataSearchUrlState: string;
+  setMetadataSearchUrlState: (metadataSearch: { metadataSearch?: string }) => void;
+}
+
+export interface KPIProps {
+  dateRange: StringDateRange;
+  dataView?: DataView;
+}
+export interface OverviewProps extends KPIProps {
+  currentTimeRange: MetricsTimeInput;
+  nodeName: string;
+  nodeType: InventoryItemType;
+  onTabsStateChange?: (tabId: TabIds) => void;
+}
+
+export const Overview = ({
+  nodeName,
+  currentTimeRange,
+  nodeType,
+  onTabsStateChange,
+  dateRange,
+  dataView,
+}: OverviewProps) => {
+  const inventoryModel = findInventoryModel(nodeType);
+  const { sourceId } = useSourceContext();
+  const {
+    loading: metadataLoading,
+    error: fetchMetadataError,
+    metadata,
+  } = useMetadata(nodeName, nodeType, inventoryModel.requiredMetrics, sourceId, currentTimeRange);
+
+  if (fetchMetadataError) {
+    return (
+      <EuiCallOut
+        title={i18n.translate('xpack.infra.metadataEmbeddable.errorTitle', {
+          defaultMessage: 'Sorry, there was an error',
+        })}
+        color="danger"
+        iconType="error"
+        data-test-subj="infraMetadataErrorCallout"
+      >
+        <FormattedMessage
+          id="xpack.infra.metadataEmbeddable.errorMessage"
+          defaultMessage="There was an error loading your data. Try to {reload} and open the host details again."
+          values={{
+            reload: (
+              <EuiLink
+                data-test-subj="infraMetadataReloadPageLink"
+                onClick={() => window.location.reload()}
```

Comment:
That was added when we implemented the metadata fetching. The idea is to reload everything if this host for example doesn't exist anymore you can't get the metadata of the same host if you reload only the tab and it will go in circles - if we reload the page we can see that the host is not in the table. 
We can think about some metadata improvements and do it both here and in the metadata component separately so it's not confusing.

Reply:
I've forced the error here, and because we store in the URL the flyout state, which contains the active `host.name`, I guess the user will continue in a loop if they reload the page


https://github.com/elastic/kibana/assets/2767137/3a082d7e-8e7d-4eed-b19d-b5b6336b5a74

I'm wondering if just showing a message here would be enough. wdyt? Besides, `window.location.reload()` forces a full page refresh

Reply:
Just realized that the Metadata tab does the same. 🤔

Reply:
> Just realized that the Metadata tab does the same. 🤔

Yes, that's why I mentioned that is better to change both places and we can do it in a separate issue and implement there only a partial reload of the flyout ( so doing the call to get the metadata ). 

> I'm wondering if just showing a message here would be enough. 

I briefly remember when we did it for the metadata tab and also other places -  the UX idea behind it was to have an action together with the errors/warnings but in general, the idea was to be a link to reload the page instead of just a text to have something the user can click directly, I think this should not be shown very often though. 

Also maybe in this case of the overview tab maybe we can customize it a bit so it's not so generic - showing the charts still (even if the metadata is not available) and then having a message that the metadata couldn't load (with try again later instead of reload link for example because we will still show something and it's not really only an error to force an action 🤔 )  wdyt?

Reply:
>I briefly remember when we did it for the metadata tab and also other places - the UX idea behind it was to have an action together with the errors/warnings but in general, the idea was to be a link to reload the page instead of just a text to have something the user can click directly, I think this should not be shown very often though.

Yeah. providing users an option to recover from a failed state is good. Loading only the metadata and not reloading the whole page might be more effective.

> Also maybe in this case of the overview tab maybe we can customize it a bit so it's not so generic - showing the charts still (even if the metadata is not available) and then having a message that the metadata couldn't load (with try again later instead of reload link for example because we will still show something and it's not really only an error to force an action 🤔 ) wdyt?

Agree! Blocking charts from loading if metadata fails to load seems weird. I like it.

Reply:
Ok, so I changed it to load the charts in the overview tab and kept the same message: 
<img width="1296" alt="image" src="https://github.com/elastic/kibana/assets/14139027/09a6f55c-d634-4c70-823c-c53ec96ea1da">

When we change the metadata reload behavior we can extract it and reuse it here ( even have a shared component for the error state ) but for now, this should be Ok.

Reply:
The new issue for the refactoring added: [#161367](https://github.com/elastic/kibana/issues/161367)